sdks/python/apache_beam/transforms/dofn_lifecycle_test.py

```diff
@@ -0,0 +1,91 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""UnitTests for DoFn lifecycle and bundle methods"""
+
+from __future__ import absolute_import
+
+import unittest
+
+from nose.plugins.attrib import attr
+
+import apache_beam as beam
+from apache_beam.testing.test_pipeline import TestPipeline
+
+class CallSequenceEnforcingDoFn(beam.DoFn):
+  def __init__(self):
+    self._setup_called = False
+    self._start_bundle_calls = 0
+    self._finish_bundle_calls = 0
+    self._teardown_called = False
+
+  def setup(self):
+    # Use assert instead of unittest.TestCase.assert* methods because
```

Comment:
You can drop the duplicate instances of this comment. (I actually think you can actually drop all instances of this comment since, but if you think it is confusing you can leave one.)

Reply:
Removed comments.